### PR TITLE
Make simulator generic

### PR DESCRIPTION
**Pull request by mads256h**
This allows us to define an Exploration simulator and a Patrolling simulator.

### CODE REVIEW

File: Assets/Scripts/Simulation/ISimulation.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        ITracker Tracker { get; }
+        
+        ISimulationInfoUIController AddSimulationInfoUIController(GameObject gameObject);
+        
+        IReadOnlyList<MonaRobot> Robots { get; }
+        
+        void SetSelectedRobot([CanBeNull] MonaRobot newSelectedRobot);
+        
+        // TODO: Remove this!
+        void SetSelectedTag([CanBeNull] VisibleTagInfoHandler newSelectedTag);
```
Comment (henneboy) [inline Assets/Scripts/Simulation/ISimulation.cs:32]: Todo in this branch or the next?

File: Assets/Scripts/Statistics/ExplorationTracker.cs [hunk 2]
```diff
@@ -167,10 +168,17 @@ public void LogicUpdate(List<MonaRobot> robots) {
             
             _currentVisualizationMode.UpdateVisualization(_explorationVisualizer, _currentTick);
             _currentTick++;
+
+            if (GlobalSettings.ShouldWriteCSVResults
```
Comment (henneboy) [inline Assets/Scripts/Statistics/ExplorationTracker.cs:172]: why did we add this?

**Review by henneboy — APPROVED** [review]
Lgtm

**Comment by henneboy** [comment]
Plz, don't move a file to a new location AND make changes to the file. Git doesn't understand so it shows up as a deletion and an addition, instead of a move and an edit :)

File: Assets/Scripts/Simulation/ISimulation.cs (hunk 1 above)
Comment (henneboy) [reply Assets/Scripts/Simulation/ISimulation.cs:32]: Next branch according to Puvi

**Review by miniJmo — APPROVED** [review]
This is a kinda big PR, so it is not the easiest to get an overview but it looks fine

File: Assets/Scripts/Statistics/ExplorationTracker.cs (hunk 2 above)
Comment (henneboy) [reply Assets/Scripts/Statistics/ExplorationTracker.cs:172]: Ok

**Pull request merged**